lib/mpl_toolkits/mplot3d/spherical.py

```diff
@@ -0,0 +1,115 @@
+                 radius=1.,
+                 subdivisions=None):
+        """
+        Parameters
+        ----------
+        vertices: an array of points in 3D assumed to be on a sphere
+        tri: a list [[i,j,k], ...] of lists [i,j,k] of integers i,j,k, where
+            i, j, k are the indices of the vertices defining the face. (If left
+            empty, class will assume only 3 vertices in clockwise order.)
+        center: a point in 3D defining the center of the sphere
+        radius: a positive double representing the radius of the sphere
+        subdivisions: an integer defining how often the triangulation of
+                      the polygon is subdivided before it is plotted. (If left
+                      empty, the class will try to guess this value such that
+                      the result looks nice.)
+        """
+
+        self.vertices = vertices
+        self.center = center
+        self.radius = radius
+        self.subdivisions = subdivisions
+        self.tri = tri
+        self._refine_triangulation()
+
+    def _project_to_sphere(self, point):
+        """
+        Projects point onto the sphere defined by self.center and self.radius.
+
+        Parameters
+        ----------
+        point : array of floats of length ndim
+
+        returns: array of floats of length ndim
+        """
+
+        length = np.linalg.norm(point - self.center)
+        return (point - self.center) / length * self.radius + self.center
+
+    def _subdivide(self):
+        """
+         Subdivides the triangulation once by replacing every triangle
+         by four smaller triangles defined by the vertices of the original
+         triangle and the midpoints of its edges.
+        """
+
+        faces_subdivided = []
+
+        for face in self.tri:
+            # create three new vertices at the midpoints of each edge
+            a = self.vertices[face[0]]
+            b = self.vertices[face[1]]
+            c = self.vertices[face[2]]
+
+            # add vertices to list
+            k = len(self.vertices)
+            self.vertices = np.insert(self.vertices, k,
+                                      self._project_to_sphere(a + b), axis=0)
+            self.vertices = np.insert(self.vertices, k + 1,
+                                      self._project_to_sphere(b + c), axis=0)
+            self.vertices = np.insert(self.vertices, k + 2,
+                                      self._project_to_sphere(a + c), axis=0)
+
+            # add new faces
+            faces_subdivided.append([face[0], k, k + 2])
+            faces_subdivided.append([k, face[1], k + 1])
+            faces_subdivided.append([k + 1, face[2], k + 2])
+            faces_subdivided.append([k, k + 1, k + 2])
+
+        self.tri = faces_subdivided
+
+    def _refine_triangulation(self):
+        """
+         Refines the triangulation of the vertices.
+        """
+
+        if self.subdivisions is None:
+            while np.min(np.array([min(
```

Comment:
I don't think the `np.min(np.array(` casting is necessary. Just take the `min` of the list comprehension.